backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -104,61 +115,88 @@ func PeriodicallyRefreshCache(s *Server) {
 	go func() {
 		for range ticker.C {
 			s.logger.Info("Refreshing RTDS cache")
-			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.logger)
-			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
 		}
 	}()
 }
 
 func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCache cacheWrapper, rtdsLayerName string,
-	logger *zap.SugaredLogger) {
-	allExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
+	ingressPrefix string, egressPrefix string, logger *zap.SugaredLogger) {
+	allRunningExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
 	if err != nil {
 		logger.Errorw("Failed to get data from experiments store", "error", err)
 
 		// If failed to get data from DB, stop all ongoing faults.
-		allExperiments = []*experimentation.Experiment{}
+		allRunningExperiments = []*experimentation.Experiment{}
 	}
 
+	// INGRESS Faults
 	// Group faults by upstream cluster
 	upstreamClusterFaultMap := make(map[string][]*experimentation.Experiment)
-	for _, experiment := range allExperiments {
+	for _, experiment := range allRunningExperiments {
```

Comment:
Would the following code allow us to get  rid of multiple for loops here?

```suggestion
        clusterFaultMap := make(map[string][]*experimentation.Experiment)
	for _, experiment := range allRunningExperiments {
		testConfig := &serverexperimentation.TestConfig{}
		if !isFaultTest(experiment, testConfig) {
			continue
		}
		
		clusterPair := testConfig.GetClusterPair()
		if testConfig.FaultInjectionType == serverexperimentation.FaultInjectionType_FAULTINJECTIONTYPE_INGRESS {
			clusterFaultMap[clusterPair.UpstreamCluster] =
				append(upstreamClusterFaultMap[clusterPair.UpstreamCluster], experiment)
		} else if testConfig.FaultInjectiontype == serverexperimentation.FaultInjectionType_FAULTINJECTIONTYPE_EGRESS {
		        clusterFaultMap[clusterPair.DownstreamCluster] =
				append(upstreamClusterFaultMap[clusterPair.DownstreamCluster], experiment)
		}
	}
```

Reply:
Great call, thanks for reviewing.